api/v1/login.go

```diff
@@ -0,0 +1,55 @@
+package v1
+
+// LoginRequest is the request for the login request.
+type LoginRequest struct {
+	Token       string             `json:"token"`
+	Network     MachineNetworkSpec `json:"network"`
+	Location    MachineLocation    `json:"location"`
+	CPUInfo     MachineCPUInfo     `json:"cpuInfo"`
+	GPUInfo     MachineGPUInfo     `json:"gpuInfo"`
+	Provider    string             `json:"provider"`
+	MachineInfo MachineInfo        `json:"machineInfo"`
+	MachineType string             `json:"machineType"`
```

Comment:
What's MachineType here? IMO gpud should not care about it.

Reply:
@cardyok ^ (removed in this PR)

Reply:
gpud does not know which type it is..